viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
 # ------------- WAVE 1 --------------------
+def create_movie(movie_title, genre, rating):
+    """
+    create a movie profile, return movie; return None if any of the values is none
+    """
+    #add new movie
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating,
+    }
+    
+    for value in new_movie.values():
+        if value == None:
+            return None
 
-def create_movie(title, genre, rating):
-    pass
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    """
+    add watched movies to "watched"
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    """
+    add movies to watch to watchlist
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    """
+    moves movie from "watch list" to "watched"
+    """
+
+    copy_of_user_data = dict(user_data)
+    movie_ready_to_move = {}
+    for i in range(len(copy_of_user_data["watchlist"])):
+        if movie_title in copy_of_user_data["watchlist"][i]["title"]:
+            movie_ready_to_move = copy_of_user_data["watchlist"][i]
+            user_data["watchlist"].remove(movie_ready_to_move)
+            user_data["watched"].append(movie_ready_to_move)
+    return user_data       
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    """
+    count average score for total rating scores in wated movie 
+    """
+
+    total_rating = 0
+    average_rating = 0
+    if user_data["watched"] != []:
+        for movie in user_data["watched"]: 
+            total_rating += movie["rating"]
+        average_rating = total_rating/len(user_data["watched"])
+    else:
+        average_rating = 0

```

Comment:
The `else` clause isn't needed here. `average_rating` will not change if the watched list is empty. 

```python
    if user_data["watched"] != []:
        for movie in user_data["watched"]: 
            total_rating += movie["rating"]
        average_rating = total_rating/len(user_data["watched"])
```